kafka/producer.go

```diff
@@ -150,11 +159,7 @@ func (ams *AsyncMessageSinkConfig) buildSaramaProducerConfig() (*sarama.Config,
 		conf.Producer.MaxMessageBytes = int(ams.MaxMessageBytes)
 	}
 
-	if ams.KeyFunc != nil {
-		conf.Producer.Partitioner = sarama.NewHashPartitioner
-	} else {
-		conf.Producer.Partitioner = sarama.NewRoundRobinPartitioner
-	}
+	conf.Producer.Partitioner = sarama.NewHashPartitioner
```

Comment:
I'm not too familiar with sarama, but I'm wondering what happens if a key is not provided by the user. Would we get unevenly distributed messages across topic partitions if we always use the hash partitioner?

Reply:
If there's no key provided to sarama but we use hash partitioner, I'd expect problems, but my proposal (which is included in my last commit) is that we use the message payload bytes as the hash input if no key was provided by the user.   That should approximate random distribution if people are not sending duplicate messages.  (And if they are doing that, the messages would also land on the same partition if they did provide the key anyway)